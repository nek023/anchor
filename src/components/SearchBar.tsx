import React from 'react'
import styled from 'styled-components'

const SearchBarInput = styled.input`
  width: 100%;
  height: 40px;
  margin: 0;
  padding: 5px;
  font-size: 26px;
  border-radius: 4px;
  border: 1px solid #cccccc;

  &:focus {
    outline: none;
  }
`

type OptionalProps = {
  onValueChanged?: (value: string) => void
}

type Props = OptionalProps & {
  value: string
}

export default class SearchBar extends React.PureComponent<Props> {
  private inputRef = React.createRef<HTMLInputElement>()

  componentDidMount() {
    if (this.inputRef.current) {
      const index = this.props.value.length
      this.inputRef.current.setSelectionRange(index, index)
    }
  }

  render() {
    return (
      <SearchBarInput
        type="text"
        value={this.props.value}
        ref={this.inputRef}
        autoFocus={true}
        onBlur={this.handleBlur}
        onChange={this.handleChange}
      />
    )
  }

  private handleBlur = () => {
    if (this.inputRef.current) {
      this.inputRef.current.focus()
    }
  }

  private handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    if (this.props.onValueChanged) {
      this.props.onValueChanged(event.target.value)
    }
  }
}
