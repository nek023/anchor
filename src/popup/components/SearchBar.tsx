import React, { useCallback, useEffect, useRef } from "react";

interface SearchBarProps {
  value: string;
  onValueChange?: (value: string) => void;
}

export const SearchBar: React.FC<SearchBarProps> = ({
  value,
  onValueChange,
}) => {
  const inputRef = useRef<HTMLInputElement>(null);

  const handleBlur = useCallback(() => inputRef?.current?.focus(), []);

  const handleChange = useCallback(
    (event: React.ChangeEvent<HTMLInputElement>) => {
      if (onValueChange == null) return;
      onValueChange(event.target.value);
    },
    [onValueChange]
  );

  useEffect(
    () => {
      // Move cursor to the end
      inputRef?.current?.setSelectionRange(value.length, value.length);
    },
    // eslint-disable-next-line react-hooks/exhaustive-deps
    []
  );

  return (
    <input
      className="w-full p-1 rounded border border-gray-300 text-2xl focus:outline-none"
      type="text"
      title="query"
      value={value}
      ref={inputRef}
      autoFocus={true}
      onBlur={handleBlur}
      onChange={handleChange}
    />
  );
};
