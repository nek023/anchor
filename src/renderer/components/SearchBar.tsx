import React, { useCallback, useEffect, useRef } from "react";
import styled from "styled-components";

const SearchBarInput = styled.input`
  width: 100%;
  height: 40px;
  margin: 0;
  padding: 5px;
  font-size: 26px;
  border-radius: 4px;
  border: 1px solid #cccccc;

  &:focus {
    outline: none;
  }
`;

interface SearchBarProps {
  value: string;
  onValueChange: (value: string) => void;
}

export const SearchBar: React.FC<SearchBarProps> = ({
  value,
  onValueChange,
}) => {
  const inputRef = useRef<HTMLInputElement>(null);

  const handleBlur = useCallback(() => inputRef?.current?.focus(), []);

  const handleChange = useCallback(
    (event: React.ChangeEvent<HTMLInputElement>) =>
      onValueChange(event.target.value),
    [onValueChange]
  );

  useEffect(
    () => inputRef?.current?.setSelectionRange(value.length, value.length),
    // eslint-disable-next-line react-hooks/exhaustive-deps
    []
  );

  return (
    <SearchBarInput
      type="text"
      value={value}
      ref={inputRef}
      autoFocus={true}
      onBlur={handleBlur}
      onChange={handleChange}
    />
  );
};
