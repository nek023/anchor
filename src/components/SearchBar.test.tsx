import { shallow } from 'enzyme'
import React from 'react'
import SearchBar from './SearchBar'

describe('SearchBar', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<SearchBar value="" />)

    expect(wrapper).toMatchSnapshot()
  })

  it('calls onValueChanged when value has changed', () => {
    const onValueChanged = jest.fn()
    const wrapper = shallow(
      <SearchBar value={'test value'} onValueChanged={onValueChanged} />
    )

    expect(onValueChanged).not.toHaveBeenCalled()

    wrapper.simulate('change', { target: { value: 'new value' } })

    expect(onValueChanged).toHaveBeenCalledTimes(1)
  })
})
