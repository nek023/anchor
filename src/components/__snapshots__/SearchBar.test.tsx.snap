// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar matches snapshot 1`] = `
<styled.input
  autoFocus={true}
  onBlur={[Function]}
  onChange={[Function]}
  type="text"
  value=""
/>
`;
